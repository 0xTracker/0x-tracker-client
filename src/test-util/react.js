import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
