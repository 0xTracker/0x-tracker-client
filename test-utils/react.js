import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRouter };
