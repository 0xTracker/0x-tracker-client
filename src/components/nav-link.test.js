import { fireEvent, cleanup } from 'react-testing-library';
import React from 'react';

import { renderWithRouter } from '../test-util/react';
import NavLink from './nav-link';

afterEach(() => {
  cleanup();
});

it('should render with url', () => {
  const { container } = renderWithRouter(
    <NavLink href="/trades/">Trades</NavLink>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('should render with active class when url matches current url', () => {
  const { container } = renderWithRouter(
    <NavLink href="/relayers">Relayers</NavLink>,
    { route: '/relayers' },
  );

  expect(container.firstChild).toHaveClass('active');
});

it('should call onClick handler when link is clicked', () => {
  const handleClick = jest.fn();
  const { getByText } = renderWithRouter(
    <NavLink onClick={handleClick}>Relayers</NavLink>,
  );

  fireEvent.click(getByText('Relayers'));

  expect(handleClick).toHaveBeenCalledTimes(1);
});
