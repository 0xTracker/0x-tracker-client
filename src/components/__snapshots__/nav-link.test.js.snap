// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with url 1`] = `
<li
  class="nav-item"
>
  <a
    class="nav-link link__StyledInternalLink-sc-1m1n17q-1 kIqcgN"
    href="/trades/"
  >
    Trades
  </a>
</li>
`;
